modules/core/src/copy.cpp

```diff
@@ -1186,9 +1186,9 @@ void cv::copyMakeBorder( InputArray _src, OutputArray _dst, int top, int bottom,
 {
     CV_INSTRUMENT_REGION();
 
-    CV_Assert( top >= 0 && bottom >= 0 && left >= 0 && right >= 0 );
+    CV_Assert( top >= 0 && bottom >= 0 && left >= 0 && right >= 0 && _src.dims() <= 2);
```

Comment:
BTW,

It is better to avoid complex assertion conditions, because they doesn't help debugging problems.
Consider using `CV_Check*()` macros instead.